src/core_modules/capture-core/components/Pages/EnrollmentAddEvent/EnrollmentAddEventPage.container.js

```diff
@@ -1,44 +1,39 @@
 // @flow
 import React, { useCallback, useMemo } from 'react';
-import { batchActions } from 'redux-batched-actions';
 // $FlowFixMe
 import { useDispatch, useSelector } from 'react-redux';
-import { useLocationQuery } from '../../../utils/routing';
-import { navigateToEnrollmentPage } from './enrollmentAddEventPage.actions';
+import { useHistory } from 'react-router-dom';
+import { buildUrlQueryString, useLocationQuery } from '../../../utils/routing';
 import { useProgramInfo } from '../../../hooks/useProgramInfo';
 import { useEnrollmentAddEventTopBar, EnrollmentAddEventTopBar } from './TopBar';
 import { EnrollmentAddEventPageComponent } from './EnrollmentAddEventPage.component';
 import { deleteEnrollment } from '../Enrollment/EnrollmentPage.actions';
 import { useWidgetDataFromStore } from './hooks';
-import {
-    useHideWidgetByRuleLocations,
-} from '../Enrollment/EnrollmentPageDefault/hooks';
+import { useHideWidgetByRuleLocations } from '../Enrollment/EnrollmentPageDefault/hooks';
 import { useCommonEnrollmentDomainData, updateEnrollmentEventsWithoutId } from '../common/EnrollmentOverviewDomain';
 import { dataEntryHasChanges as getDataEntryHasChanges } from '../../DataEntry/common/dataEntryHasChanges';
 
 export const EnrollmentAddEventPage = () => {
     const { programId, stageId, orgUnitId, teiId, enrollmentId } = useLocationQuery();
-
+    const history = useHistory();
     const dispatch = useDispatch();
 
     const handleCancel = useCallback(() => {
-        dispatch(navigateToEnrollmentPage(programId, orgUnitId, teiId, enrollmentId));
-    }, [dispatch, programId, orgUnitId, teiId, enrollmentId]);
+        history.push(`/enrollment?${buildUrlQueryString({ orgUnitId, programId, teiId, enrollmentId })}`);
```

Comment:
is there a reason why don't we keep the `dispatch` and modify the `navigateToEnrollmentPage` function?

Reply:
I think my reasoning for this change is that we do not have access to the useHistory-hook in `EnrollmentAddEventPage.actions`, and therefore it would be easier to just remove it. We could pass in the history object, but this would require some rewrite. Does this make sense, or is it something I'm forgetting? 
Thanks!